packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -0,0 +1,473 @@
+        childWhenDragging: const SizedBox(),
+        dragAnchor: DragAnchor.child,
+        onDragStarted: onDragStarted,
+        // When the drag ends inside a DragTarget widget, the drag
+        // succeeds, and we swap the widget into position appropriately.
+        onDragCompleted: onDragEnded,
+        // When the drag does not end inside a DragTarget widget, the
+        // drag fails, but we still swap the widget to the last position it
+        // had been dragged to.
+        onDraggableCanceled: (Velocity velocity, Offset offset) {
+          onDragEnded();
+        },
+      );
+
+      // The target for dropping at the end of the list doesn't need to be
+      // draggable.
+      if (index >= widget.children.length) {
+        child = toWrap;
+      }
+
+      // Determine the size of the drop area to show under the dragging widget.
+      final Widget spacing = (widget.scrollDirection == Axis.vertical)
+          ? new SizedBox(height: _dropAreaExtent)
+          : new SizedBox(width: _dropAreaExtent);
+
+      // We open up a space under where the dragging widget currently is to
+      // show it can be dropped.
+      if (_currentIndex == index) {
+        return _buildContainerForScrollDirection(children: <Widget>[
+          new SizeTransition(
+            sizeFactor: _entranceController, 
+            axis: widget.scrollDirection,
+            child: spacing
+          ),
+          child,
+        ]);
+      }
+      // We close up the space under where the dragging widget previously was
+      // with the ghostController animation.
+      if (_ghostIndex == index) {
+        return _buildContainerForScrollDirection(children: <Widget>[
+          new SizeTransition(
+            sizeFactor: _ghostController, 
+            axis: widget.scrollDirection,
+            child: spacing,
+          ),
+          child,
+        ]);
+      }
+      return child;
+    }
+
+    // We wrap the drag target in a Builder so that we can scroll to its specific context.
+    return new Builder(builder: (BuildContext context) {
+      return new DragTarget<Key>(
+        builder: buildDragTarget,
+        onWillAccept: (Key toAccept) {
+          setState(() {
+            _nextIndex = index;
+            _requestAnimationToNextIndex();
+          });
+          _scrollTo(context);
+          // If the target is not the original starting point, then we will accept the drop.
+          return _dragging == toAccept && toAccept != toWrap.key;
+        },
+        onAccept: (Key accepted) {},
+        onLeave: (Key leaving) {},
+      );
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    // We use the layout builder to constrain the cross-axis size of dragging child widgets.
+    return new LayoutBuilder(builder: (BuildContext context, BoxConstraints constraints) {
+        final List<Widget> wrappedChildren = <Widget>[];
+        for (int i=0; i<widget.children.length; i++) {
```

Comment:
spaces and i += 1: for (int i = 0; i  < widget.children.length; i += 1)

Reply:
Done